File: src/main/java/de/shevchuk/urlshortener/repository/UrlRepository.java
package de.shevchuk.urlshortener.repository;

import de.shevchuk.urlshortener.entity.Url;
import java.util.List;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UrlRepository extends CrudRepository<Url, String> {
    //overridden otherwise the default return Iterable
    List<Url> findAll();
}

File: src/main/java/de/shevchuk/urlshortener/repository/UrlRepository.kt
package de.shevchuk.urlshortener.repository

import de.shevchuk.urlshortener.entity.Url
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface UrlRepository : CrudRepository<Url, String> {
    //overridden otherwise the default return Iterable
    override fun findAll(): List<Url>
}